includes/admin/wc-meta-box-functions.php

```diff
@@ -181,6 +181,26 @@ function woocommerce_wp_checkbox( $field ) {
 	echo '</p>';
 }
 
+
+/**
+ * Return true if stringified $value is found in array of stringified $options or if stringified $value
+ * is the same as stringified $options.
+ *
+ * @param string|int       $value   Value to find within options.
+ * @param string|int|array $options Options to go through when looking for value.
+ * @return bool
+ */
+function wc_is_value_in_options( $value, $options ) {
```

Comment:
👍 

Since this is specialised in nature how about we replace `selected` itself. I.e. create `wc_selected` which applies your new logic and returns a string - `selected` if true, empty if false. That takes care of markup too and should be cleaner overall.

A unit test for that helper function would also be useful.

Reply:
+1 for unit test for this function Unit tests are a good way of documenting the correct behavior so we don't accidentally break this in the future. 

Since this is a bug fix it can be merged before or after the 3.4 beta is tagged, so you have time.